pvlib/clearsky/nrel.py

```diff
@@ -0,0 +1,187 @@
++/-10% with rigourous radiative transfer codes. The model computes solar
+radiation for every hour of the year, based on the 10 user input parameters.
+
+The graphical presentation includes diurnal clear sky radiation patterns for
+every day of the year. The user may copy cells of interest or the graph and
+paste it to an unprotected worksheet for manipulation.
+
+The workbook is PROTECTED using the password BIRD (all caps). To generate data
+for the entire year, choose TOOLS, PROTECTION, UNPROTECT and enter the password.
+Copy row 49 and paste it from row 50 all the way down to row 8761. 
+
+NOTE: Columns L to U contain intermediate calculations and have been collapsed
+down for convenient pressentation of model results.
+
+Contact:
+Daryl R. Myers,
+National Renewable Energy Laboratory,
+1617 Cole Blvd. MS 3411, Golden CO 80401
+(303)-384-6768 e-mail daryl_myers@nrel.gov
+
+http://rredc.nrel.gov/solar/models/clearsky/
+http://rredc.nrel.gov/solar/pubs/pdfs/tr-642-761.pdf
+http://rredc.nrel.gov/solar/models/clearsky/error_reports.html
+"""
+
+import numpy as np
+import pandas as pd
+import seaborn as sns
+import xlrd
+import os
+
+
+def bird(doy, hr, lat, lon, tz, press_mB, o3_cm, h2o_cm, aod_500nm, aod_380nm,
+         b_a, alb, lyear=365.0, solar_constant=1367.0):
+    """
+    Bird Simple Clearsky Model
+
+    :param doy: day(s) of the year
+    :type doy: int
+    :param hr: hour
+    :param lat: latitude [degrees]
+    :param lon: longitude [degrees]
+    :param tz: time zone
+    :param press_mB: pressure [mBar]
+    :param o3_cm: atmospheric ozone [cm]
+    :param h2o_cm: precipital water [cm]
+    :param aod_500nm: aerosol optical depth [cm] measured at 500[nm]
+    :param aod_380nm: aerosol optical depth [cm] measured at 380[nm]
+    :param b_a: asymmetry factor
+    :param alb: albedo 
+    """
+    doy0 = doy - 1.0
+    patm = 1013.0
+    day_angle = 6.283185 * doy0 / lyear
+    # rad2deg = 180.0 / np.pi
+    dec_rad = (
+        0.006918 - 0.399912 * np.cos(day_angle) + 0.070257 * np.sin(day_angle) -
+        0.006758 * np.cos(2.0 * day_angle) +
+        0.000907 * np.sin(2.0 * day_angle) -
+        0.002697 * np.cos(3.0 * day_angle) + 0.00148 * np.sin(3.0 * day_angle)
+    )
+    declination = np.rad2deg(dec_rad)
+    # equation of time
+    eqt = 229.18 * (
+        0.0000075 + 0.001868 * np.cos(day_angle) -
+        0.032077 * np.sin(day_angle) - 0.014615 * np.cos(2.0 * day_angle) -
+        0.040849 * np.sin(2.0 * day_angle)
+    )
+    hour_angle = 15.0 * (hr - 12.5) + lon - tz * 15.0 + eqt / 4.0
+    lat_rad = np.deg2rad(lat)
+    ze_rad = np.arccos(
+        np.cos(dec_rad) * np.cos(lat_rad) * np.cos(np.deg2rad(hour_angle)) +
+        np.sin(dec_rad) * np.sin(lat_rad)
+    )
+    zenith = np.rad2deg(ze_rad)
+    airmass = np.where(zenith < 89,
+        1.0 / (np.cos(ze_rad) + 0.15 / (93.885 - zenith) ** 1.25), 0.0
+    )
+    pstar = press_mB / patm
+    am_press = pstar*airmass
+    t_rayliegh = np.where(airmass > 0,
```

Comment:
not necessary anymore, since relative airmass is `NaN` if not valid
:heavy_check_mark: